podpac/core/cache/cache_ctrl.py

```diff
@@ -0,0 +1,264 @@
+        '''
+        
+        if not isinstance(node, podpac.Node):
+            raise TypeError("node must of type 'Node', not '%s'" % type(Node))
+
+        if not isinstance(key, six.string_types):
+            raise TypeError("key must be a string type, not '%s'" % (type(key)))
+        
+        if not isinstance(coordinates, podpac.Coordinates) and coordinates is not None:
+            raise TypeError("coordinates must be of type 'Coordinates', not '%s'" % type(coordinates))
+
+        if not isinstance(mode, six.string_types) and mode is not None:
+            raise TypeError("mode must be of type 'str', not '%s'" % type(mode))
+
+        if key == '*':
+            raise ValueError("key cannot be '*'")
+
+        for c in self._get_cache_stores(mode):
+            if c.has(node=node, key=key, coordinates=coordinates):
+                    return True
+        
+        return False
+
+    def rem(self, node, key, coordinates=None, mode=None):
+        '''Delete cached data for this node.
+        
+        Parameters
+        ----------
+        node : Node, str
+            node requesting storage.
+        key : str
+            Delete only cached objects with this key. Use `'*'` to match all keys.
+        coordinates : Coordinates, str
+            Delete only cached objects for these coordinates. Use `'*'` to match all coordinates.
+        mode : str
+            determines what types of the `CacheStore` are affected: 'ram','disk','network','all'. Defaults to `node._cache_mode` or 'all'. Overriden by `self._cache_mode` if `self._cache_mode` is not `None`.
+        '''
+
+        if not isinstance(node, podpac.Node):
+            raise TypeError("node must of type 'Node', not '%s'" % type(Node))
+
+        if not isinstance(key, six.string_types):
+            raise TypeError("key must be a string type, not '%s'" % (type(key)))
+        
+        if not isinstance(coordinates, podpac.Coordinates) and coordinates is not None and coordinates != '*':
+            raise TypeError("coordinates must be '*' or of type 'Coordinates' not '%s'" % type(coordinates))
+
+        if not isinstance(mode, six.string_types) and mode is not None:
+            raise TypeError("mode must be of type 'str', not '%s'" % type(mode))
+
+        if key == '*':
+            key = CacheWildCard()
+
+        if coordinates == '*':
+            coordinates = CacheWildCard()
+
+        for c in self._get_cache_stores(mode):
+            c.rem(node=node, key=key, coordinates=coordinates)
+
+    def clear(self, mode=None):
+        """
+        Clear all cached data.
+
+        Parameters
+        ------------
+        mode : str
+            determines what types of the `CacheStore` are affected: 'ram','disk','network','all'. Defaults to `node._cache_mode` or 'all'. Overriden by `self._cache_mode` if `self._cache_mode` is not `None`.
+        """
+
+        if not isinstance(node, podpac.Node):
```

Comment:
I think `node` is undefined here (in `cache_ctrl.clear()`).

Reply:
Yep, thanks. Need those cache_ctrl tests.

Reply:
I'm working on a commit that fixes this one.

Reply:
thanks guys. I'll keep looking, and comment if I see anything else. This is big rewrite. I appreciate the restructuring into multiple files, `cache.py` was getting too big.